### DISCUSSION TITLE
Question!What is the difference between e3nn and torch_geometric

### DISCUSSION THREAD

**Discussion by Silencezong** [e3nn/e3nn#319] [Q&A] [answered]
Hello everyone, I just came into e3nn, and I want to know what is the difference and connection between e3nn and graph neural networks(PYG). Going deeper, what is the difference between e3nn and se3transoformer

**Comment by mariogeiger** [comment, marked as answer]
- pyg allows to implement graph neural networks
- e3nn and pyg together allows to implement **equivariant** graph neural networks
- se3transoformer [official implementation](https://github.com/FabianFuchsML/se3-transformer-public) is using an [old version of e3nn](https://github.com/FabianFuchsML/se3-transformer-public/tree/master/equivariant_attention/from_se3cnn). Look [here](https://docs.e3nn.org/en/stable/guide/transformer.html#transformer-guide) for a more modular implementation using e3nn.

**Reply by Silencezong** [reply]
thanks bro！
